src/mlpack/methods/edge_boxes/feature_extraction.hpp

```diff
@@ -0,0 +1,262 @@
+  
+  /**
+   * Computes distance transform of 1D vector f.
+   * @param f input vector whose distance transform is to be found.
+   * @param n size of the Output vector to be made.
+   * @param inf a large double value.
+   * @param d Output vector which stores distance transform of f.
+   */
+  void DistanceTransform1D(const arma::vec& f, const size_t n,
+                           const double inf, arma::vec& d);
+  
+  /**
+   * Computes distance transform of a 2D array
+   * @param Im input array whose distance transform is to be found.
+   * @param inf a large double value.
+   */
+  void DistanceTransform2D(MatType &Im, const double inf);
+
+  /**
+   * euclidean distance transform of binary Image using squared distance
+   * This is the discription of the paper which discribes the approach 
+   * for this algorithm : Distance Transforms of Sampled Functions,
+   * P. Felzenszwalb, D. Huttenlocher
+   * Theory of Computing, Vol. 8, No. 19, September 2012
+   * @param Im Input binary Image whose distance transform is to be found.
+   * @param on if on == 1, 1 is taken as boundaries and vice versa.
+   * @param Out Output Image.
+   */  
+  void DistanceTransformImage(const MatType& Im, double on, MatType& Out);
+
+  /**
+   * Compute the Regular and Self Similarity Features of the given image.
+   * @param Image Given Input Image
+   * @param loc Locations at which features need to be extracted
+   * @param RegFtr Output the Regular Features
+   * @param SSFtr Output the Self Similarity Features
+   * @param table a helper vector required to convert image to LAB space.
+   */
+  void GetFeatures(const MatType &Image, arma::umat &loc,
+                   CubeType& RegFtr, CubeType& SSFtr,
+                   const arma::vec& table);
+
+  /**
+   * Makes a reflective border around an Image.
+   * @param InImage, Image which we have to make border around.
+   * @param top, border length (to be incremented) at top.
+   * @param left, border length at left.
+   * @param bottom, border length at bottom.
+   * @param right, border length at right.
+   * @param OutImage, Output Image. 
+   */  
+  void CopyMakeBorder(const CubeType& InImage, size_t top, 
+                 size_t left, size_t bottom, size_t right,
+                 CubeType& OutImage);
+  
+  /**
+   * Augment image patch with multiple channels of information 
+   * resulting in a feature vector. Shrink these channels by shrink
+   * size to reduce dimensions of the extracted candidate features.
+   * Refer to the following paper for details:
+   * Fast edge detection using structured forests
+   * Authors: Piotr Dollar, Larry Zitnick
+   * Published In: ICCV
+   * @param InImage Input Image.
+   * @param regCh Channels used in calculating Regular Features.
+   * @param ssCh Channels used in calculating Self Similarity Features.
+   * @param table a vector which helps in converting image from RGB to LUV.
+   */
+  void GetShrunkChannels(const CubeType& InImage, CubeType &reg_ch,
```

Comment:
Can you use the doxygen style to cite the paper. Here is an example: https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/layer/dropout_layer.hpp